compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaSingletonTypeSymbol.java

```diff
@@ -53,6 +55,10 @@ public BallerinaSingletonTypeSymbol(CompilerContext context, BLangLiteral shape,
         } else {
             this.typeName = shape.toString();
         }
+
+        // Getting the original type of the shape
+        TypesFactory typesFactory = TypesFactory.getInstance(this.context);
+        originalType = typesFactory.getTypeDescriptor(shape.getBType());
```

Comment:
It is not good to create a type-factory instance inside the constructor. Why don't we pass the original type from the outside?